### ISSUE TITLE
Unknown variable: '::snmp_agentaddress' error

### ISSUE THREAD

**Issue by jdehnert** [voxpupuli/puppet-snmp#65]
I'm running your snmp module on a puppet 4 puppetserver (V 2.2.1) on Centos 6.7, and I'm getting the following error when I run:

`Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Unknown variable: '::snmp_agentaddress'. at /etc/puppetlabs/code/environments/test/modules/snmp/manifests/params.pp:19:19 on node it-lnx-01.enphaseenergy.com`

I have created the following in my config files:

```
class { 'snmp':
  ro_community => 'password',
}
```

and 

```
node /^it-lnx-\d+$/ {
  include common
  include mcollective::server
  include ::snmp
}
```

Any ideas what I'm doing wrong?

 Thanks,
    James "Zeke" Dehnert
## 

mailto:zdehnert@enphase.com  James "Zeke" Dehnert
                  -= Eschew Obfuscation =-
     "Life is racing. Everything else is just waiting"

**Comment by jimmihagen** [comment]
This error message indicates the variable was never initialized. The error goes away if the strict_variables setting is not 'true', which will let you use this module as-is, but could potentially introduce errors.

**Comment by mterzo** [comment]
Upgraded to puppet 4 and this module has tons of these issues. I'm going to submit a patch I don't see anyone else working on this.

**Comment by jnquintin** [comment]
Corrected in the pr#79
